frontend/src/BoardDetailPage.jsx

```diff
@@ -1,24 +1,40 @@
-import { use } from "react";
 import { useParams } from "react-router-dom"; //going to use this to get whatever params got passed into route (board id so we can render right info)
-import { cards, boards } from "./data";
+import { useState } from "react";
+import { useEffect } from "react";
+
+import { getBoardWithCards } from "../utils";
+import { deleteCard } from "../utils";
 import "./boarddetailpage.css";
 import Header from "./components/Header/Header";
 import Footer from "./components/Footer/Footer";
 import PostList from "./components/PostList/PostList";
 
 export default function BoardDetailPage() {
   const { id } = useParams(); //get the id of the board we just clicked on whos posts we want to view. This id is accesible through the route since it is dynamic and each has their own link
-  const parsedId = parseInt(id); //parse the id since it is returned as a string. We need to do this for comparison below
+  const [boardPosts, setBoardPosts] = useState([]);
+  const [board, setBoard] = useState([]);
+
+  async function handleDelete(cardId) {
+    const deletedCard = await deleteCard(cardId, board.id);
+    const updatedCards = boardPosts.filter(
+      (post) => post.cardId !== deletedCard.cardId
+    );
+    setBoardPosts(updatedCards);
```

Comment:
Is better directly do the operation here (assuming you don't change it with the fetch):
```suggestion
    setBoardPosts(prevBoardPosts => prevBoardPosts.filter(
      (post) => post.cardId !== deletedCard.cardId
    ));
```